tests/utils/test_databricks_utils.py

```diff
@@ -102,21 +102,49 @@ def test_databricks_single_slash_in_uri_scheme_throws(get_config):
         databricks_utils.get_databricks_host_creds("databricks:/profile:path")
 
 
-def test_databricks_model_serving_throws(monkeypatch):
-    monkeypatch.setenv("DATABRICKS_MODEL_SERVING_ENV", "true")
-    monkeypatch.setenv("DATABRICKS_MODEL_SERVING_HOST_URL", "host")
+def test_get_model_dependency_token(oauth_file):
+    with mock.patch(
+        "mlflow.utils.databricks_utils._MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH", str(oauth_file)
+    ):
+        token = databricks_utils.get_model_dependency_oauth_token()
+        assert token == "token2"
+
+
+def test_get_model_dependency_oauth_token_model_serving_throws():
     with pytest.raises(MlflowException, match="Unable to read Oauth credentials"):
-        databricks_utils.get_databricks_host_creds()
+        databricks_utils.get_model_dependency_oauth_token()
 
 
-def test_databricks_params_model_serving_oauth_cache(monkeypatch):
-    monkeypatch.setenv("DATABRICKS_MODEL_SERVING_ENV", "true")
+def test_databricks_params_model_serving_oauth_cache(monkeypatch, oauth_file):
+    monkeypatch.setenv("IS_IN_DATABRICKS_MODEL_SERVING_ENV", "true")
     monkeypatch.setenv("DATABRICKS_MODEL_SERVING_HOST_URL", "host")
     monkeypatch.setenv("DATABRICKS_DEPENDENCY_OAUTH_CACHE", "token")
     monkeypatch.setenv("DATABRICKS_DEPENDENCY_OAUTH_CACHE_EXIRY_TS", str(time.time() + 5))
-    params = databricks_utils.get_databricks_host_creds()
-    assert params.host == "host"
-    assert params.token == "token"
+    # oauth file still needs to be present for should_fetch_model_serving_environment_oauth()
+    #  to evaluate true
+    with mock.patch(
+        "mlflow.utils.databricks_utils._MODEL_DEPENDENCY_OAUTH_TOKEN_FILE_PATH", str(oauth_file)
+    ):
+        params = databricks_utils.get_databricks_host_creds()
+        assert params.host == "host"
+        # should use token from cache, rather than token from oauthfile
+        assert params.token == "token"
+
+
+def test_databricks_params_env_var_overrides_model_serving_oauth(monkeypatch, oauth_file):
+    monkeypatch.setenv("IS_IN_DATABRICKS_MODEL_SERVING_ENV", "true")
+    monkeypatch.setenv("DATABRICKS_MODEL_SERVING_HOST_URL", "host")
+    monkeypatch.setenv("DATABRICKS_HOST", "host_envvar")
+    monkeypatch.setenv("DATABRICKS_TOKEN", "pat_token")
+    # oauth file still needs to be present for should_fetch_model_serving_environment_oauth()
```

Comment:
Shouldn't we be setting

```
    monkeypatch.setenv("DATABRICKS_DEPENDENCY_OAUTH_CACHE", "token")
    monkeypatch.setenv("DATABRICKS_DEPENDENCY_OAUTH_CACHE_EXIRY_TS", str(time.time() + 5))
```

Here too?

Reply:
it's not neccesary, if env var doesn't exist it will try to read from oauthfile